### DISCUSSION TITLE
Help needed Git diff output does not look similar to how the text looks in neovim

### DISCUSSION THREAD

**Discussion by aliasgar55** [neovim/neovim#28648] [General]
Hi

I am writing golang and the code in neovim does not look the same as the output in git diff command
The aligning of code looks proper in neovim but when the same code is viewed in git diff the tabs/spaces in the front does not have the correct spacing

I have attached screenshots 

```Git diff output```
![image](https://github.com/neovim/neovim/assets/67861700/60215fc1-2190-44d5-a3d5-157bd485781c)

```neovim output```
![image](https://github.com/neovim/neovim/assets/67861700/40a38406-51cc-4982-848f-3cc31018681e)

**Comment by aliasgar55** [comment]
In the above screenshots the health_follow_redirects line spacing looks as if its aligned properly in the neovim but the same output is not shown in the `git diff` command

**Comment by zeertzjq** [comment]
It's possible that Tab size is 8 in Nvim, while it is 4 in `git diff`, which causes the different appearance.

**Reply by aliasgar55** [reply]
```vim.opt.guicursor = ""

vim.opt.nu = true
vim.opt.relativenumber = true

vim.opt.tabstop = 4
vim.opt.softtabstop = 4
vim.opt.shiftwidth = 4
vim.opt.expandtab = true

vim.opt.smartindent = true

vim.opt.wrap = false

vim.opt.swapfile = false
vim.opt.backup = false
vim.opt.undodir = os.getenv("HOME") .. "/.vim/undodir"
vim.opt.undofile = true

vim.opt.hlsearch = false
vim.opt.incsearch = true

vim.opt.scrolloff = 8
vim.opt.signcolumn = "yes"
vim.opt.isfname:append("@-@")

vim.opt.updatetime = 50

vim.opt.colorcolumn = "80"



vim.g.mapleader = " "
```

**Reply by zeertzjq** [reply]
Hmm, try taking a screenshot after `:set list`.

**Reply by aliasgar55** [reply]
it looks something like this
![image](https://github.com/neovim/neovim/assets/67861700/5e987996-b5c7-4561-876b-75d9d570fc41)

**Reply by aliasgar55** [reply]
removing the line `vim.opt.tabstop = 4` fixes the issue

**Reply by aliasgar55** [reply]
Turning expandtab off fixes the issue.
